src/BodyParams/BodyParamsMiddleware.php

```diff
@@ -82,10 +83,17 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
             }
 
             // Matched! Parse and pass on to the next
-            return $next(
-                $strategy->parse($request),
-                $response
-            );
+            try {
+                return $next(
+                    $strategy->parse($request),
+                    $response
+                );
+            } catch (MalformedRequestBodyException $exception) {
```

Comment:
@weierophinney I considered that but thought it might be better to give the developer the option of further handling of the request.

What if they wanted to change / translate the `$reasonPhrase`? Or log the error? Or any number of other things? Calling the next middleware would give them that opportunity; however, for those who don't need to do anything additional, it puts the burden of checking for a malformed request and cutting the pipe on them.

Although, I suppose if they needed special handling after the fact, they could extend / compose the `BodyParamsMiddleware` and deal with it like that...

Given that, do you still think it makes more sense to return the response? I'm fine with it either way, just wanted to make we took everything into consideration.